src/psyclone/psyir/transformations/nemo_sign_trans.py

```diff
@@ -0,0 +1,183 @@
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: R. W. Ford, STFC Daresbury Lab
+
+'''Module providing a NEMO-API-specific transformation from a PSyIR
+SIGN operator to PSyIR code. This could be useful if the SIGN operator
+is not supported by the back-end or if the performance of the inline
+code is better than the intrinsic.
+
+The implementation is NEMO-specific as NEMO code generation does not
+currently create a symbol table, see issue #500. Once this has been
+implemented the transformation can be modified to work for all APIs.
+
+'''
+from psyclone.undoredo import Memento
+from psyclone.psyir.transformations.nemo_operator_trans import \
+    NemoOperatorTrans
+from psyclone.psyir.transformations import NemoAbsTrans
+from psyclone.psyir.nodes import UnaryOperation, BinaryOperation, Assignment, \
+    Reference, Literal, IfBlock
+from psyclone.psyir.symbols import DataType, DataSymbol
+
+
+class NemoSignTrans(NemoOperatorTrans):
+    '''Provides a NEMO-API-specific transformation from a PSyIR SIGN
+    Operator node to equivalent code in a PSyIR tree. Validity checks
+    are also performed.
+
+    The transformation replaces `R=SIGN(A,B)` with the following logic:
+
+    `R=SIGN(A,B) => R=ABS(A); if B<0.0 R=R*-1.0`
+
+    i.e. the value of A with the sign of B
+
+    Note, an alternative implementation would be:
+    `if B<0 then (if A<0 R=A else R=A*-1) else ((if A>0 R=A else R=A*-1))`
+
+    '''
+    def __init__(self):
+        super(NemoSignTrans, self).__init__()
+        self._operator_name = "SIGN"
+        self._classes = (BinaryOperation,)
+        self._operators = (BinaryOperation.Operator.SIGN,)
+
+    def apply(self, node, symbol_table, options=None):
+        '''Apply the SIGN intrinsic conversion transformation to the specified
+        node. This node must be an SIGN BinaryOperation. The SIGN
+        BinaryOperation is converted to the following equivalent
+        inline code:
+
+        R=SIGN(A,B) => R=ABS(A); if B<0.0 R=R*-1.0
+
+        This is implemented as a transform from:
+
+        R= ... SIGN(A,B) ...
+
+        to:
+
+        tmp_abs=A
```

Comment:
Again, please try and see whether use of rst markup can improve the appearance of the generated doc.

Reply:
Improved for all three transformations.